### PR TITLE
Add modern icons following new Jenkins UI style

### PR DESCRIPTION
**Pull request by DuMaM**
I'm using latest Jenkins weekly release. It starts to hurt my eyes when i see those old style icons.
I'm not a graphic designer, but I created them by combining some already made form those sites:
https://github.com/jenkinsci/jenkins/tree/master/war/src/main/resources/images/symbols
https://www.svgrepo.com/

I'm not sure about buildbadge.svg and confighistory.svg, but others I think are ok.

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Review by NotMyFault — COMMENTED** [review]
I agree, the Tango icons are by far no longer up to date. Jenkins core uses [ionicons](https://ionic.io/ionicons) as modern symbols, we probably should do the same here too.

Can you move these icons over to use the [symbol API](https://weekly.ci.jenkins.io/design-library/Symbols/)?
Otherwise symbol don't adapt other themes and retain their original color:
Bildschirmfoto 2022-05-25 um 09 42 21

**Comment by DuMaM** [comment]
@NotMyFault could you take a look for it now? 
I'm still checking but i do not have experience with jelly so im not sure if those changes are legit.

File: src/main/java/hudson/plugins/jobConfigHistory/JobConfigHistoryConsts.java [hunk 1]
```diff
@@ -37,7 +37,7 @@ public final class JobConfigHistoryConsts {
     /**
      * Path to the icon.
      */
-    public static final String ICONFILENAME = "/plugin/jobConfigHistory/img/confighistory.svg";
+    public static final String ICONFILENAME = "/plugin/jobConfigHistory/images/sources/confighistory.svg";
```
Comment (NotMyFault) [inline src/main/java/hudson/plugins/jobConfigHistory/JobConfigHistoryConsts.java:40]:
```suggestion
    public static final String ICONFILENAME = "symbol-confighistory plugin-jobConfigHistory";

```

**Review by NotMyFault — COMMENTED** [review]
Looks promising so far, I'll give it a practical test later.

File: src/main/java/hudson/plugins/jobConfigHistory/JobConfigBadgeAction.java [hunk 2]
```diff
@@ -139,7 +139,7 @@ public String getTooltip() {
      * @return Icon path as string
      */
     public String getIcon() {
-        return "/plugin/jobConfigHistory/img/buildbadge.svg";
+        return "/plugin/jobConfigHistory/images/symbols/buildbadge.svg";
```
Comment (DuMaM) [inline src/main/java/hudson/plugins/jobConfigHistory/JobConfigBadgeAction.java:142]:
```suggestion
        return "symbol-buildbadge plugin-jobConfigHistory";
```

**Comment by DuMaM** [comment]
Ok, your suggestion probably fix loading config and badge. 😉 
I stuck there for some time.
I also will change background document symbol to match `All configs` one.
There is also strange issue, with overflowing two groups. I don't know why 😕 
In preview there is no overlapping for `filter.svg` symbol.
image

**Comment by DuMaM** [comment]
image
image
image
image
image
Ok it seems to work 🎉, but icons require some tweaks 😭

**Comment by NotMyFault** [comment]
Not too sure about that one, but the oversized icons need a fixed viewbox in these locations, because Jenkins doesn't attempt to resize oversized icons here.

**Comment by DuMaM** [comment]
@NotMyFault I think i fixed all issues ;)

File: src/main/resources/hudson/plugins/jobConfigHistory/JobConfigBadgeAction/badge.jelly [hunk 3]
```diff
@@ -1,8 +1,8 @@
 <?jelly escape-by-default='true'?>
-<j:jelly xmlns:j="jelly:core">
+<j:jelly xmlns:j="jelly:core" xmlns:l="/lib/layout" >
   <j:if test="${it.showBadge() and it.oldConfigsExist()}">
-    <a href="${it.createLink()}" id="showDiff">
-      <img width="16" height="16" title="${it.getTooltip()}" alt="x" src="${resURL}${it.getIcon()}"/>
-    </a>
+    <span class="icon-sm" >
+      <l:icon id="showDiff" src="${it.getIcon()}" href="${it.createLink()}" title="${it.getTooltip()}" alt="x" />
+    </span>
```
Comment (NotMyFault) [inline src/main/resources/hudson/plugins/jobConfigHistory/JobConfigBadgeAction/badge.jelly:6]: This doesn't quite work out, I can't click the icon, how about
```xml
<a href="${it.createLink()}">
    <span class="icon-sm">
      <l:icon src="${it.getIcon()}" title="${it.getTooltip()}" />
    </span>
</a>
```

**Comment by DuMaM** [comment]
Nice catch. I didn't noticed that hyperlinks do not work.

File: src/main/resources/hudson/plugins/jobConfigHistory/JobConfigBadgeAction/badge.jelly (hunk 3 above)
Comment (DuMaM) [reply src/main/resources/hudson/plugins/jobConfigHistory/JobConfigBadgeAction/badge.jelly]: done

**Comment by DuMaM** [comment]
@NotMyFault I know why this wont work.
It's do to new symbol api. It do not allow for href and tooltips...

https://github.com/jenkinsci/jenkins/blob/2d62032ca422c332b8324ff8925444bd4c9562f9/core/src/main/resources/lib/layout/icon.jelly#L67

**Comment by DuMaM** [comment]
Everything now is working fine 😉 
At first i didn't noticed some functionality. Thanks @NotMyFault for suggestions.
image
image
image
image

I'm only wondering why i got an underscore before badge😕

**Comment by NotMyFault** [comment]
Bildschirmfoto 2022-05-26 um 10 05 45

It looks like core itself adds a nbsp there, which the a href makes visible, because it's now a clickable URL.

**Comment by DuMaM** [comment]
Are you aware of any href style I should use to remove this underline?

**Comment by NotMyFault** [comment]
That'll work for now.

**Comment by DuMaM** [comment]
@NotMyFault Should we also add this to 'restore' buttons. They are now highlighted with blue.

**Comment by NotMyFault** [comment]
> @NotMyFault Should we also add this to 'restore' buttons. They are now highlighted with blue.

Yeah, possibly.

**Review by NotMyFault — APPROVED** [review]

File: pom.xml [hunk 4]
```diff
@@ -13,7 +13,6 @@
     <version>${changelist}</version>
 
     <name>Jenkins Job Configuration History Plugin</name>
-    <description>Saves copies of job, node and system configurations.</description>
```
Comment (DuMaM) [inline pom.xml:16]: @NotMyFault why did you remove this line?

Comment (NotMyFault) [reply pom.xml:16]: To comply with the standards of an upcoming hard requirement: https://github.com/jenkinsci/maven-hpi-plugin/blob/master/src/main/java/org/jenkinsci/maven/plugins/hpi/HpiMojo.java#L139
